uploadservice/src/main/java/net/gotev/uploadservice/NotificationAction.java

```diff
@@ -28,8 +30,12 @@ public String getText() {
         return text;
     }
 
-    public PendingIntent getClickIntent() {
-        return clickIntent;
+    public final PendingIntent getPendingIntent(Context context) {
+        if (clickIntent == null) {
+            return PendingIntent.getBroadcast(context, 0, new Intent(), PendingIntent.FLAG_UPDATE_CURRENT);
```

Comment:
This way you could only have broadcast pending intents, so if you want to open an activity, you can't. I think the implementation was better before this last commit.
I agree that to create an `Intent` is less code than a `PendingIntent`, but you loose flexibility this way.